### DISCUSSION TITLE
Faulty compiler error message CS0165

### DISCUSSION THREAD

**Discussion by gafter** [dotnet/csharplang#2267] [Language Ideas]
@vsfeedback commented on [Mon Feb 25 2019](https://github.com/dotnet/roslyn/issues/33671)

A compiler error  message CS0165 appears even though the code is correct.

 private static void MicrosoftFehler()
 {
      MsFehler msf = new MsFehler();

      //  It works - no message appears
      if (msf != null &amp;&amp;  msf.Fkt(out int n) == true)
      {
        Console.WriteLine(n);
      }

      // Here a message appears, although the code is correct:
      // Error CS0165 Use of unassigned local variable  &quot;m&quot;
      //------------------------------------------------------------------------------
      if (msf?.Fkt(out int m) == true)
      {
---&gt;        Console.WriteLine(m);                &lt;----------------------------------------------------------------------------
      }

    }

    
    class MsFehler
    {
      public bool Fkt(out int i)
      {
        i = 100;

        return true;
      }
    }

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/47334/faulty-compiler-error-message-cs0165.html
VSTS ticketId: 421730_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

---

@gafter commented on [Tue Feb 26 2019](https://github.com/dotnet/roslyn/issues/33671#issuecomment-467566782)

Although I have sympathy, the current language specification does not have separate "assigned when true" and "assigned when false" states for `operator==`.  Therefore this is a language feature request and I am moving it to csharplang.

**Comment by Unknown6656** [comment]
@gafter : you can use GitHub's issue transferal tool for that ^^.

**Comment by Joe4evr** [comment]
I already told him in #2260.

**Comment by yaakov-h** [comment]
Duplicate of https://github.com/dotnet/csharplang/issues/1195?

**Comment by gafter** [comment]
> you can use GitHub's issue transferal tool for that

No I cannot because I am not an admin for this repository.

**Comment by Unknown6656** [comment]
Oh my bad, I thought that `[Member]`s have the same permissions.

**Comment by jnm2** [comment]
@Unknown6656 Please send feedback to GitHub about this. I haven't been able to use the issue transfer feature yet at NUnit either for the same reason, and it would be really useful there as well.
